srtcore/channel.cpp

```diff
@@ -460,47 +499,27 @@ EReadStatus CChannel::recvfrom(sockaddr* addr, CPacket& packet) const
     DWORD flag = 0;
     int addrsize = m_iSockAddrSize;
 
-    int res = ::WSARecvFrom(m_iSocket, (LPWSABUF)packet.m_PacketVector, 2, &size, &flag, addr, &addrsize, NULL, NULL);
-    res = (0 == res) ? size : -1;
+    int sockerror = ::WSARecvFrom(m_iSocket, (LPWSABUF)packet.m_PacketVector, 2, &size, &flag, addr, &addrsize, NULL, NULL);
+	int res = (0 == sockerror) ? size : -1;
+	if (sockerror)
+	{
+		// On Windows this is a little bit more complicated, so simply treat every error
+		// as an "again" situation. This should still be probably fixed, but it needs more
+		// thorough research. For example, the problem usually reported from here is
+		// WSAETIMEDOUT, which isn't mentioned in the documentation of WSARecvFrom at all.
+		int err = NET_ERROR;
+		LOGC(mglog.Debug) << CONID() << "(sys)WSARecvFrom: " << SysStrError(err) << " [" << err << "]";
+		status = RST_AGAIN;
```

Comment:
What will happen if network goes away, ex. cable gets unplugged?

Reply:
This is completely explained in the initial description. In case when the network goes down, the reader will retry reading indefinitely - which is how the older version were behaving (with the difference that stransmit on Windows also doesn't implement the data passing timeout feature correctly due to nonexistent `alarm()`). Of course, this should be fixed, but the fix that is needed for Windows here is much bigger and deserves a more effort consuming activity (everything is explained in comments around it).